File: check_tags.py

import re

def check_html_tags(content):
    # Very simple tag checker
    tags = re.findall(r'<(/?[a-zA-Z0-9]+)', content)
    stack = []
    for tag in tags:
        if tag.startswith('/'):
            if not stack:
                print(f"Balanise fermante inattendue: {tag}")
                continue
            last = stack.pop()
            if last != tag[1:]:
                print(f"Erreur de correspondance: ouvert {last}, fermé {tag}")
        else:
            # Self-closing tags in HTML templates (simplified list)
            if tag in ['img', 'br', 'hr', 'input', 'meta', 'link']:
                continue
            stack.append(tag)
    
    if stack:
        print(f"Balises non fermées: {stack}")
    else:
        print("Toutes les balises semblent équilibrées.")

with open(r'c:\Users\HARMONY\Desktop\portfolioarchi-main\src\app\pages\parcours\parcours.ts', 'r', encoding='utf-8') as f:
    content = f.read()
    # Extract template content
    template_match = re.search(r'template: `(.*?)`', content, re.DOTALL)
    if template_match:
        check_html_tags(template_match.group(1))
    else:
        print("Template non trouvé")

File: check_tags.js

const fs = require('fs');

function checkHtmlTags(content) {
    const tagRegex = /<(\/?[a-zA-Z0-9]+)/g;
    let match;
    const stack = [];
    while ((match = tagRegex.exec(content)) !== null) {
        const tag = match[1];
        if (tag.startsWith('/')) {
            if (stack.length === 0) {
                console.log(`Unexpected closing tag: ${tag}`);
                continue;
            }
            const last = stack.pop();
            if (last !== tag.substring(1)) {
                console.log(`Mismatched tags: opened ${last}, closed ${tag}`);
            }
        } else {
            // Simplified self-closing list
            if (['img', 'br', 'hr', 'input', 'meta', 'link'].includes(tag.toLowerCase())) {
                continue;
            }
            stack.push(tag);
        }
    }
    
    if (stack.length > 0) {
        console.log(`Unclosed tags: ${stack.join(', ')}`);
    } else {
        console.log("Tags seem balanced.");
    }
}

const content = fs.readFileSync('c:\\Users\\HARMONY\\Desktop\\portfolioarchi-main\\src\\app\\pages\\parcours\\parcours.ts', 'utf8');
const templateMatch = content.match(/template:\s*`(.*?)`/s);
if (templateMatch) {
    checkHtmlTags(templateMatch[1]);
} else {
    console.log("Template not found");
}
